datrie/tail.c

```diff
@@ -251,6 +251,57 @@ tail_fwrite (const Tail *t, FILE *file)
 }
 
 
+size_t
+get_tail_serialized_size (const Tail *t)
+{
+    size_t static_count = (
+        sizeof(int32) * 3 // TAIL_SIGNATURE , t->first_free , t->num_tails
+    );
+    size_t dynamic_count = 0u;
+    if(t->num_tails > 0){
+        dynamic_count += (
+            4 * 2 // t->tails[i].next_free , t->tails[i].data
+            + 2 // length
+        ) * t->num_tails;
+        for (TrieIndex   i = 0; i < t->num_tails; i++) {
+            if (t->tails[i].suffix > 0)
+            {
+                dynamic_count += strlen ((const char *)t->tails[i].suffix) * sizeof(const char);
+            }
+        }
+    }
+    return static_count + dynamic_count;
+}
+
+
+int
+tail_serialize (const Tail *t, unsigned char **ptr)
+{
+    TrieIndex   i;
+
+    serialize_int32_be_incr (ptr, TAIL_SIGNATURE);
+    serialize_int32_be_incr (ptr, t->first_free);
+    serialize_int32_be_incr (ptr, t->num_tails);
+
+    for (i = 0; i < t->num_tails; i++) {
+        int16   length;
+        serialize_int32_be_incr (ptr, t->tails[i].next_free);
+        serialize_int32_be_incr (ptr, t->tails[i].data);
+
+        length = t->tails[i].suffix ? strlen ((const char *)t->tails[i].suffix)
+                                    : 0;
```

Comment:
Same as above, `strlen()` -> `trie_char_strlen()`.

Reply:
IDK why isn't it marked as `Outdated`, fixed it too: https://github.com/tlwg/libdatrie/pull/12/files#diff-07e263274dea1ce3a38a600987ed672b9326336cf5799bcd71138ea580c667e5R294-R295